python/unblob/report.py

```diff
@@ -287,18 +312,49 @@ def log_with(self, logger):
         )
 
 
-@attrs.define(kw_only=True, frozen=True)
-class LinkExtractionProblem(ExtractionProblem):
+class LinkExtractionProblem(ExtractionProblemBase):
     link_path: str
+    report_type: Literal["LinkExtractionProblem"] = "LinkExtractionProblem"
 
     def log_with(self, logger):
         logger.warning(self.log_msg, path=self.path, link_path=self.link_path)
 
 
-@attrs.define(kw_only=True, frozen=True)
-class SpecialFileExtractionProblem(ExtractionProblem):
+class SpecialFileExtractionProblem(ExtractionProblemBase):
     mode: int
     device: int
+    report_type: Literal["SpecialFileExtractionProblem"] = (
+        "SpecialFileExtractionProblem"
+    )
```

Comment:
You probably need to adapt typing for `record_problem` in `python/unblob/file_utils`:

```
  /home/runner/work/unblob/unblob/python/unblob/file_utils.py:573:33 - error: Argument of type "SpecialFileExtractionProblem" cannot be assigned to parameter "problem" of type "ExtractionProblem" in function "record_problem"
    "SpecialFileExtractionProblem" is not assignable to "ExtractionProblem" (reportArgumentType)
```